### ISSUE TITLE
System.NullReferenceException on trace 1.3

### ISSUE THREAD

**Issue by daninden** [DataDog/dd-trace-dotnet#376]
**Describe the bug**
After updating from version 1.0 to version 1.3 of the trace agent (both msi as nuget packages are updated) because of a StackoverflowException on Azure Storage the AspNetCoreMvc2Integration is giving errors on my MVC calls.

**System.NullReferenceException**: Object reference not set to an instance of an object.
   at System.Object.GetType()
   at Datadog.Trace.ClrProfiler.Emit.ObjectExtensions.TryGetPropertyValue[TResult](Object source, String propertyName, TResult& value)
   at Datadog.Trace.ClrProfiler.Integrations.AspNetCoreMvc2Integration.Rethrow(Object context, Int32 opCode)

**System.NullReferenceException**: Object reference not set to an instance of an object.
   at System.Object.GetType()
   at Datadog.Trace.ClrProfiler.Integrations.Interception.ParamsToTypes(Object[] objectsToCheck)
   at Datadog.Trace.ClrProfiler.Integrations.AspNetCoreMvc2Integration.Rethrow(Object context, Int32 opCode)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.ResultNext[TFilter,TFilterAsync](State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAlwaysRunResultFilters()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeFilterPipelineAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()
   at Microsoft.AspNetCore.Builder.RouterMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Builder.Extensions.MapWhenMiddleware.Invoke(HttpContext context)

**To Reproduce**
Any call on our test environment is producing this.

**Expected behavior**
No errors and working application

**Screenshots**
![image](https://user-images.githubusercontent.com/7497671/59029114-b619f000-885d-11e9-9d8a-9eec60950383.png)

- Instrumentation mode: automatic with msi installer
- Tracer version: 1.3.0
- OS: Windows Server 2012 R2 
- CLR: .NET Core 2.2

**Additional context**
I'm running my asp.net core application as a Windows Service.

**Comment by andrewsouthard1** [comment]
Thanks for flagging @daninden our team is going to take a look into this.

**Comment by cwoolum** [comment]
Having the same problem. This has completely crippled our dev environment.

``` csharp

System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Object.GetType()
   at Datadog.Trace.ClrProfiler.Integrations.Interception.ParamsToTypes(Object[] objectsToCheck)
   at Datadog.Trace.ClrProfiler.Integrations.AspNetCoreMvc2Integration.Rethrow(Object context)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.ResultNext[TFilter,TFilterAsync](State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeResultFilters()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeFilterPipelineAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.Invoke(HttpContext httpContext)
   at Test.Architecture.Api.Core.Middleware.ErrorLoggingMiddleware.Invoke(HttpContext context)
```

**Comment by lucaspimentel** [comment]
Thanks again for reporting this issue. It should be resolved with [release 1.3.1](https://github.com/DataDog/dd-trace-dotnet/releases/tag/v1.3.1).

**Comment by daninden** [comment]
I can confirm that it's fixed now
